woocommerce/woocommerce-android
# [Coupons] Edit amount

Closes: #5536 


### Description
This PR adds support for editing the coupon's amount, and the list of changes is:
1. Add a `hasChanges` property to the ViewModel, and update the Save button depending on its value.
2. Update the amount unit depending on the coupon type.
3. Add some overloads for WCOutlinedTextField to support more advanced scenarios, and a specific component version that allows typed values.
4. Handle the amount edition.

### Testing instructions
1. Make sure the coupons beta toggle is enabled.
2. Open the app, then click on More menu, then coupons.
3. Click on one of the coupons.
4. Click on Menu, then select "edit".
5. Confirm the "Save" button is disabled.
6. Update the content of the amount field.
7. Make sure to test the field properly: handling of cursor position, allowed characters, backspace behavior...
8. Confirm that the "Save" button gets enabled when the amount is different than what the coupon had.

### Images/gif
https://user-images.githubusercontent.com/1657201/167250737-fa40579b-081e-4aab-95eb-02ce540a24de.mov
